This is a wdspec test.
[FAIL] test_params_request_invalid_type[None]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_request_invalid_type[False]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_request_invalid_type[42]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_request_invalid_type[value3]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_request_invalid_type[value4]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_request_invalid_value[]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_request_invalid_value[foo]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_request_no_such_request
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_reason_phrase_invalid_type[False]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_reason_phrase_invalid_type[42]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_reason_phrase_invalid_type[value2]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_reason_phrase_invalid_type[value3]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_status_code_invalid_type[False]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_status_code_invalid_type[foo]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_status_code_invalid_type[value2]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_status_code_invalid_type[value3]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_status_code_invalid_value[-1]
  AttributeError: 'Network' object has no attribute 'provide_response'
[FAIL] test_params_status_code_invalid_value[4.3]
  AttributeError: 'Network' object has no attribute 'provide_response'
Harness: the test ran to completion.
